app/models/cart.rb

```diff
@@ -32,6 +32,15 @@ def grand_total
     grand_total
   end
 
+  def discounted_grand_total
+    grand_total = 0.0
+    @contents.each do |item_id, quantity|
+      grand_total += discounted_subtotal_of(item_id) if find_discount(item_id)
+      grand_total += subtotal_of(item_id) if !find_discount(item_id)
+    end
```

Comment:
The only reason I made this a separate method is because I wanted to display a different message in the cart depending on if the total has been discounted or not.